pkg/controller/node/node_controller.go

```diff
@@ -109,7 +116,18 @@ func (n *NodeController) SyncHandler(key string) error {
 			cloudproviderapi.TaintExternalCloudProvider)
 		return nil
 	}
-	nodeEgressIPConfigs, err := n.cloudProviderClient.GetNodeEgressIPConfiguration(node)
+	cloudPrivateIPConfigs, err := n.cloudPrivateIPConfigLister.List(labels.Everything())
+	if err != nil {
+		return fmt.Errorf("error listing cloud private ip config, err: %v", err)
+	}
+	index := 0
+	for _, cloudPrivateIPConfig := range cloudPrivateIPConfigs {
+		if isAssignedCloudPrivateIPConfigOnNode(cloudPrivateIPConfig, key) {
+			cloudPrivateIPConfigs[index] = cloudPrivateIPConfig
+			index++
+		}
+	}
+	nodeEgressIPConfigs, err := n.cloudProviderClient.GetNodeEgressIPConfiguration(node, cloudPrivateIPConfigs[:index])
```

Comment:
Oh, that took me a while:
You're just reusing the cloudPrivateIPConfigs slice, overwriting the lower keys, and then passing a slice with the index's length. Perhaps document what you're doing with a comment?  (but perhaps it's just me, that's really optional)